packages/core/src/button/Button.tsx

```diff
@@ -72,15 +89,47 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
           withBaseName(variant),
           {
             [withBaseName("disabled")]: disabled,
+            [withBaseName(`loading-${variant}`)]: isLoading,
             [withBaseName("active")]: active,
           },
           className
         )}
+        aria-disabled={disabled}
+        aria-live={isLoading !== undefined ? "assertive" : undefined}
         {...restProps}
         ref={ref}
         type={type}
       >
-        {children}
+        {isLoading ? (
+          <>
+            <span className={clsx(withBaseName("loading-overlay"), className)}>
+              <Spinner
+                size="small"
+                className={clsx(withBaseName("loading-spinner"), className)}
+              />
+              <span
```

Comment:
Even if this is the right handling of accessibility, why is this used over the built-in controls for Spinner?